### PR TITLE
LLT-5547: Use generated adapter type

### PR DESCRIPTION
**Pull request by mathiaspeters**
This is the final PR migrating from bespoke types in natlab to types from generated python bindings. This PR consists of almost only changing from the old `AdapterType` (defined in `telio.py`)  to `TelioAdapterType` which is in the generated bindings. 

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

File: nat-lab/tests/telio.py [hunk 1]
```diff
@@ -1158,11 +1142,3 @@ async def collect_core_dumps(self):
                     f" {container_name}:{file_path} {core_dump_destination}"
                 )
                 os.system(cmd)
-
-
-def generate_secret_key() -> str:
```
Comment (tomasz-grz) [inline nat-lab/tests/telio.py:1163]: Why were these removed?

File: nat-lab/tests/test_mesh_plus_vpn.py [hunk 2]
```diff
@@ -268,37 +269,39 @@ async def test_mesh_plus_vpn_both_peers(
...
             ConnectionTag.WINDOWS_VM_1,
-            AdapterType.WireguardGo,
+            TelioAdapterType.WIREGUARD_GO_TUN,
             "10.0.254.7",
             marks=pytest.mark.windows,
         ),
         pytest.param(
             ConnectionTag.MAC_VM,
-            AdapterType.Default,
+            None,
```
Comment (tomasz-grz) [inline nat-lab/tests/test_mesh_plus_vpn.py:295]: What happens if the adapter is set to `None`? Is the default one used for each platform? AFAIK only BoringTun works on macOS

**Review by tomasz-grz — COMMENTED** [review]
Nice work +1.0, just a few questions from my side

File: nat-lab/tests/telio.py (hunk 1 above)
Comment (mathiaspeters) [reply nat-lab/tests/telio.py:1163]: All this function is doing is calling a function in the bindings, so instead of having this function, the call sites can just call that function from the bindings directly

File: nat-lab/tests/test_mesh_plus_vpn.py (hunk 2 above)
Comment (mathiaspeters) [reply nat-lab/tests/test_mesh_plus_vpn.py:295]: The logic for that is handled here: https://github.com/NordSecurity/libtelio/pull/839/files#diff-f1122f50402db2f1613c75eb1db5aed7bc4c9ed907b99f1bf4e5010345bfe6cdR793 and there is a comment for the `SetupParams` class here https://github.com/NordSecurity/libtelio/pull/839/files#diff-9fa2bd5404cf7ee9aeb4e031cc531e8d22f729d73f6891181350bfab552fa181R43 that I updated. Basically, `None` tells natlab to use the default adapter for each platform which is `WireguardGo` on windows and `BoringTun` on all other platforms

File: nat-lab/tests/telio.py [hunk 3]
```diff
@@ -813,6 +790,13 @@ def get_features(self) -> Features:
         assert self._telio_features
         return self._telio_features
 
+    def get_adapter_type(self) -> TelioAdapterType:
+        if self._adapter_type is not None:
+            return self._adapter_type
+        if isinstance(self.get_router(), WindowsRouter):
+            return TelioAdapterType.WIREGUARD_GO_TUN
+        return TelioAdapterType.BORING_TUN
```
Comment (stalowyjez) [inline nat-lab/tests/telio.py:798]: I guess these two are written BoringTun and WireGuard, so why one of them is written with underscore and the other one without?

**Review by stalowyjez — COMMENTED** [review]
Looks good, +1

File: nat-lab/tests/telio.py (hunk 3 above)
Comment (mathiaspeters) [reply nat-lab/tests/telio.py:798]: If you check in the `.udl` it's `WireguardGoTun` and `BoringTun`, hence `WIREGUARD_GO_TUN` and `BORING_TUN`

File: nat-lab/tests/test_mesh_plus_vpn.py (hunk 2 above)
Comment (tomasz-grz) [reply nat-lab/tests/test_mesh_plus_vpn.py:295]: In that sense the `AdapterType.Default` pattern was a little bit more descriptive of what is actually going on.
What do you think about naming the parameter `override_adapter_type` instead then?

**Review by tomasz-grz — DISMISSED** [review]

**Review by tomasz-grz — APPROVED** [review]
Nice 🥇 +1
